### PR TITLE
zip: refresh

### PR DESCRIPTION
**Pull request by git-em**
Expand options, add basic invocation (without opts), rm -s option

- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**

### CODE REVIEW

**Comment by git-em** [comment]
See https://github.com/tldr-pages/tldr/pull/7825#issuecomment-1053651249

File: pages/common/zip.md [hunk 1]
```diff
@@ -3,34 +3,34 @@
...
+
+- Exclude unwanted files from being added to the compressed archive:
+
+`zip --exclude {{path/to/exclude}} --recurse-paths {{compressed.zip}} {{path/to/directory}}`
 
 - Add files to an existing zip file:
 
 `zip {{compressed.zip}} {{path/to/file}}`
 
 - Delete files from an existing zip file:
```
Comment (blueskyson) [inline pages/common/zip.md:30]: This is more precise.
```suggestion
- Delete entries from an existing zip file:
```

File: pages/common/zip.md [hunk 2]
```diff
@@ -3,34 +3,34 @@
...
 
-`zip -d {{compressed.zip}} "{{foo/*.tmp}}"`
-
-- Archive a directory and its contents to a multi-part [s]plit zip file (e.g. 3 GB parts):
-
-`zip -r -s {{3g}} {{compressed.zip}} {{path/to/directory}}`
+`zip --delete {{compressed.zip}} "{{foo/*.tmp}}"`
```
Comment (blueskyson) [inline pages/common/zip.md:32]: Provide 3 useful syntax.
```suggestion
`zip --delete {{compressed.zip}} "{{foo/file}}" "{{foo/bar/*}}" "{{*.ext}}"`
```

**Comment by CleanMachine1** [comment]
Hi, thanks for your contribution, could you please change the syntax of the things like {{path/to/file1}} to be {{[path/to/file_1}}

**Review by CleanMachine1 — APPROVED** [review]

File: pages/common/zip.md [hunk 3]
```diff
@@ -3,34 +3,34 @@
...
 - Create an encrypted archive (user will be prompted for a password):
 
-`zip -e -r {{compressed.zip}} {{path/to/directory}}`
+`zip --encrypt --recurse-paths {{compressed.zip}} {{path/to/directory}}`
 
-- Add files to an existing zip file:
+- Exclude unwanted files from being added to the compressed archive:
 
-`zip {{compressed.zip}} {{path/to/file}}`
+`zip --exclude {{path/to/exclude}} --recurse-paths {{compressed.zip}} {{path/to/directory}}`
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:24]:
```suggestion
`zip --exclude {{path/to/file1 path/to/file2 ...}} --recurse-paths {{compressed.zip}} {{path/to/directory}}`
```

File: pages/common/zip.md [hunk 4]
```diff
@@ -3,34 +3,34 @@
...
 
-- Delete files from an existing zip file:
+- Add files to an existing zip file:
 
-`zip -d {{compressed.zip}} "{{foo/*.tmp}}"`
+`zip {{compressed.zip}} {{path/to/file}}`
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:28]:
```suggestion
`zip {{path/to/compressed.zip}} {{path/to/file1 path/to/file2 ...}}`
```

File: pages/common/zip.md [hunk 5]
```diff
@@ -3,34 +3,34 @@
...
 
-- Archive a directory and its contents to a multi-part [s]plit zip file (e.g. 3 GB parts):
+- Delete entries from an existing zip file:
 
-`zip -r -s {{3g}} {{compressed.zip}} {{path/to/directory}}`
+`zip --delete {{compressed.zip}} "{{foo/file}}" "{{foo/bar/*}}" "{{*.ext}}"`
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:32]:
```suggestion
`zip --delete {{path/to/compressed.zip}} {{path/to/file_to_remove1 path/to/file_to_remove2 ...}}`
```

File: pages/common/zip.md [hunk 6]
```diff
@@ -3,34 +3,34 @@
...
 
 - List files within a specified archive (without extracting them):
 
-`zip -sf {{compressed.zip}}`
+`zip --show-files {{compressed.zip}}`
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:36]:
```suggestion
`zip --show-files {{path/to/compressed.zip}}`
```

**Review by EmilySeville7cfg — CHANGES_REQUESTED** [review]

File: pages/common/zip.md (hunk 3 above)
Comment (navarroaxel) [reply pages/common/zip.md:24]: I think the current example is better.

File: pages/common/zip.md [hunk 7]
```diff
@@ -3,34 +3,34 @@
 > Package and compress (archive) files into zip file.
 > More information: <https://manned.org/zip>.
 
-- Package and compress files and directories [r]ecursively:
+- Package and compress files:
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:6]:
```suggestion
- Package and compress the specified files:
```

File: pages/common/zip.md [hunk 8]
```diff
@@ -3,34 +3,34 @@
...
 
-`zip -r {{compressed.zip}} {{path/to/file}} {{path/to/directory1}} {{path/to/directory2}}`
+`zip {{path/to/compressed.zip}} {{path/to/file1 path/to/file2 ...}}`
 
-- E[x]clude unwanted files from being added to the compressed archive:
+- Package and compress files and directories recursively:
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:10]:
```suggestion
- Package and compress the specified files and directories recursively:
```

File: pages/common/zip.md [hunk 9]
```diff
@@ -3,34 +3,34 @@
...
 
-`zip -r {{compressed.zip}} {{path/to/directory}} -x {{path/to/exclude}}`
+`zip --recurse-paths {{path/to/compressed.zip}} {{path/to/file_or_directory1 path/to/file_or_directory2 ...}}`
 
 - Archive a directory and its contents with the highest level [9] of compression:
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:14]:
```suggestion
- Archive the specified directory with the highest level (9) of compression:
```

File: pages/common/zip.md [hunk 10]
```diff
@@ -3,34 +3,34 @@
...
 
-`zip -r -{{9}} {{compressed.zip}} {{path/to/directory}}`
+`zip -{{9}} --recurse-paths {{path/to/compressed.zip}} {{path/to/directory}}`
 
 - Create an encrypted archive (user will be prompted for a password):
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:18]:
```suggestion
- Create the encrypted archive (user will be prompted for the password):
```

File: pages/common/zip.md [hunk 11]
```diff
@@ -3,34 +3,34 @@
...
+`zip --encrypt --recurse-paths {{path/to/compressed.zip}} {{path/to/directory}}`
 
-- Add files to an existing zip file:
+- Exclude unwanted files from being added to the compressed archive:
 
-`zip {{compressed.zip}} {{path/to/file}}`
+`zip --exclude {{path/to/exclude}} --recurse-paths {{compressed.zip}} {{path/to/directory}}`
 
-- Delete files from an existing zip file:
+- Add files to an existing zip file:
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:26]:
```suggestion
- Add the specified files to the existing zip file:
```

File: pages/common/zip.md [hunk 12]
```diff
@@ -3,34 +3,34 @@
...
 
-`zip -d {{compressed.zip}} "{{foo/*.tmp}}"`
+`zip {{path/to/compressed.zip}} {{path/to/file1 path/to/file2 ...}}`
 
-- Archive a directory and its contents to a multi-part [s]plit zip file (e.g. 3 GB parts):
+- Delete entries from an existing zip file:
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:30]:
```suggestion
- Delete the specified entries from the existing zip file:
```

File: pages/common/zip.md [hunk 13]
```diff
@@ -3,34 +3,34 @@
...
 
-`zip -r -s {{3g}} {{compressed.zip}} {{path/to/directory}}`
+`zip --delete {{path/to/compressed.zip}} {{path/to/file_to_remove1 path/to/file_to_remove2 ...}}`
 
 - List files within a specified archive (without extracting them):
```
Comment (EmilySeville7cfg) [inline pages/common/zip.md:34]:
```suggestion
- List files within the specified archive (without extracting them):
```

**Review by EmilySeville7cfg — CHANGES_REQUESTED** [review]

**Comment by mfrw** [comment]
Looks like there is one more PR for refreshing zip:
- #7996 

What should we do with this PR ?

**Comment by EmilySeville7cfg** [comment]
@mfrw, I think we can merge them (take my changes and apply them to this PR where needed). I can close my PR.

**Comment by mfrw** [comment]
> @mfrw, I think we can merge them (take my changes and apply them to this PR where needed). I can close my PR.

@EmilySeville7cfg In my opinion, the PR #7996 seems more complete.
Probably we should reopen that and port changes from here.
/cc @git-em @navarroaxel

**Comment by CleanMachine1** [comment]
closing due to inactivity, open again if you would still like to contribute this.

**Pull request closed without merging**